File: JAVA/L3/P1/Lesson.java
package LABS.L3.P1;

import java.util.Date;
import java.util.Scanner;

public class Lesson {
    private Date lessonDate;
    private Timetable timetable;

    public Lesson(Date lessonDate, Timetable timetable) {
        this.lessonDate = lessonDate;
        this.timetable=timetable;
    }

    public Date getLessonDateDate() {
        return lessonDate;
    }

    public void setLessonDate(Date lessonDate) {
        this.lessonDate = lessonDate;
    }

    public Timetable getTimetable() {
        return timetable;
    }

    public void setTimetable(Timetable timetable) {
        this.timetable = timetable;
    }

    public void fill(Scanner scanner){
        lessonDate = new Date(scanner.nextInt());
        timetable.fill(scanner);
    }

    @Override
    public String toString() {
        return "Lesson:" +
                "Date= " + lessonDate +
                "Timetable= " + timetable;
    }
}

File: C#/L5/Lesson.cs
namespace L5
{
    public class Lesson : InOut
    {
        private DateTime lessonDate;
        private Timetable timetable;

        public Lesson()
        {
            lessonDate = new DateTime(0);
            timetable = new Timetable();
        }

        public Lesson(DateTime lessonDate, Timetable timetable)
        {
            this.lessonDate = lessonDate;
            this.timetable = timetable;
        }

        public DateTime getLessonDateDate()
        {
            return lessonDate;
        }

        public void setLessonDate(DateTime lessonDate)
        {
            this.lessonDate = lessonDate;
        }

        public Timetable getTimetable()
        {
            return timetable;
        }

        public void setTimetable(Timetable timetable)
        {
            this.timetable = timetable;
        }

        public void fill()
        {
            Console.WriteLine("\nLesson\nlessonDate(DateTime):\n");
            if (!DateTime.TryParseExact(Console.ReadLine(), "dd.MM.yyyy", null, System.Globalization.DateTimeStyles.None, out lessonDate))
                lessonDate = DateTime.Now;
            timetable.fill();
        }

        public void toString()
        {
            Console.Write("\nLesson:\n" +
                    "Date = " + lessonDate.ToString("dd.MM.yyyy") + "\n");
                
            timetable.toString();
        }
    }
}
